btrdb/stream.py

```diff
@@ -1145,6 +1151,26 @@ def windows(self, width, depth):
         specified. This is much faster to execute on the database side.
 
         """
+        start, end = None, None
+        if len(args) < 2 and len(kwargs) < 2:
+            raise UserWarning("Require 'width' and 'depth'")
+        elif len(args) == 2 and len(kwargs) == 0:
+            width, depth = args
+        elif len(args) == 0 and len(kwargs) == 2:
+            width, depth = kwargs.get('width', None), kwargs.get('depth', None)
+        elif len(args) == 0 and len(kwargs) == 4:
+            start, end, width, depth = [kwargs.get(k, None) for k in ('start', 'end', 'width',  'depth')]
+        elif len(args) == 4 and len(kwargs) == 0:
```

Comment:
I think overall overall that having multiple ways to order & submit the input args could add more confusion than usability.